src/main/java/uk/gov/hmcts/ccd/domain/model/search/elasticsearch/ElasticsearchRequest.java

```diff
@@ -59,22 +93,64 @@ public JsonNode getSource() {
                 .map(JsonNode::asText)
                 .map(this::getFieldId)
                 .filter(Objects::nonNull)
-                .collect(Collectors.toList());
+                .collect(toList());
         }
 
         return Collections.emptyList();
     }
 
     public boolean hasQuery() {
-        return searchRequest.has(QUERY);
+        return nativeSearchRequest.has(QUERY);
     }
 
     public JsonNode getQuery() {
-        return searchRequest.get(QUERY);
+        return nativeSearchRequest.get(QUERY);
+    }
+
+    public boolean hasSupplementaryData() {
+        return supplementaryData != null && supplementaryData.isArray();
```

Comment:
rather than adding checks in multiple places checking if supp data is an array, we could validate that once at the controller level where we check the validity of the request, and then assume it's an array here